pkg/actuators/machine/reconciler.go

```diff
@@ -446,3 +475,237 @@ func (r *Reconciler) setMachineAddresses(instance *models.PVMInstance) error {
+func (r *Reconciler) updateLoadBalancers(internalIP string) error {
+	if len(r.providerSpec.LoadBalancers) == 0 {
+		klog.V(3).Infof("%s: Instance has no load balancers configured. Skipping", r.machine.Name)
+		return nil
+	}
+	var applicationLoadBalancerNames []string
+	for _, loadBalancerRef := range r.providerSpec.LoadBalancers {
+		switch loadBalancerRef.Type {
+		case machinev1.ApplicationLoadBalancerType:
+			applicationLoadBalancerNames = append(applicationLoadBalancerNames, loadBalancerRef.Name)
+		}
+	}
+	if len(applicationLoadBalancerNames) > 0 {
+		if err := registerWithApplicationLoadBalancers(r.powerVSClient, applicationLoadBalancerNames, internalIP); err != nil {
+			klog.Errorf("%s: Failed to register application load balancers: %v", r.machine.Name, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func registerWithApplicationLoadBalancers(powerVSClient client.Client, loadBalancerNames []string, internalIP string) error {
+	lbMap, err := getLoadBalancers(powerVSClient, loadBalancerNames)
+	if err != nil {
+		return err
+	}
+	return createLoadBalancerPoolMember(powerVSClient, lbMap, internalIP)
+}
+
+type loadBalanacerOptions struct {
+	id    string
+	pools []vpcv1.LoadBalancerPoolReference
+	state string
+}
+
+func getLoadBalancers(powerVSClient client.Client, loadBalancerNames []string) (map[string]loadBalanacerOptions, error) {
+	// construct the map from provided LoadBalancer names for easy search
+	loadBalancerNamesMap := make(map[string]struct{})
+	for _, lb := range loadBalancerNames {
+		loadBalancerNamesMap[lb] = struct{}{}
+	}
+
+	var loadBalancersList []vpcv1.LoadBalancer
+	f := func(start string) (bool, string, error) {
+		listLoadBalancersOptions := &vpcv1.ListLoadBalancersOptions{}
+		if start != "" {
+			listLoadBalancersOptions.Start = &start
+		}
+
+		loadBalancers, _, err := powerVSClient.ListLoadBalancers(listLoadBalancersOptions)
+		if err != nil {
+			return false, "", err
+		}
+		if loadBalancers != nil {
+			loadBalancersList = append(loadBalancersList, loadBalancers.LoadBalancers...)
+			if loadBalancers.Next != nil && *loadBalancers.Next.Href != "" {
+				return false, *loadBalancers.Next.Href, nil
+			}
+		}
+		return true, "", nil
+	}
+
+	if err := utils.PagingHelper(f); err != nil {
+		return nil, fmt.Errorf("error listing loadbalancer %v", err)
+	}
+
+	if loadBalancersList == nil {
+		return nil, errors.New("no loadbalancer is retrieved")
+	}
+
+	// construct LoadBalancer name with LoadBalancer ID map
+	lbMap := make(map[string]loadBalanacerOptions)
+	for _, lb := range loadBalancersList {
+		if _, ok := loadBalancerNamesMap[*lb.Name]; ok {
+			lbMap[*lb.Name] = loadBalanacerOptions{
+				id:    *lb.ID,
+				pools: lb.Pools,
+				state: *lb.ProvisioningStatus,
+			}
+		}
+	}
+	if len(lbMap) != len(loadBalancerNames) {
+		return nil, fmt.Errorf("not able to find %s loadbalancer in cloud", loadBalancerNames)
```

Comment:
There is a possibility that this will print load balancers that are exist too in case of more than one load balancer is on search.